nx.json

```diff
@@ -17,51 +17,92 @@
     }
   },
   "namedInputs": {
-    "global": [
-      "{workspaceRoot}/yarn.lock",
+    "workspace": [
       "{workspaceRoot}/package.json",
+      "{workspaceRoot}/nx.json",
+      "{workspaceRoot}/.nxignore",
+      "{workspaceRoot}/tools/@o3r/workspace-helpers/scripts/*.mjs",
+      "{workspaceRoot}/jest.config.js"
+    ],
+    "project": [
+      "{projectRoot}/package.json",
+      "{projectRoot}/project.json"
+    ],
+    "base": [
+      "project",
+      "{workspaceRoot}/yarn.lock",
```

Comment:
should we put `"${workspaceRoot}/package.json"` ?

Reply:
in fact, why isn't workspace part of base?

Reply:
I'm just confused why we have `yarn.lock` but not `package.json`

Reply:
In case of any dependency update the `yarn.lock` is updated but any update of the `package.json` does not mean a dependency update.
I don't see why `package.json` would trigger any project cache clean that the `yarn.lock` would not :S

Reply:
An upgrade of the version of yarn perhaps ? This would be covered by the `package.json` but not the `yarn.lock`, probably
That's the only thing I can think of, but I'm not even sure we want to invalidate the cache for that

Reply:
I would say we do not want to invalidate the cache for it, the yarn upgrade should not after the build result (or we have a very serious issue :D)